### PR TITLE
TINKERPOP-1550 Made ganglia/graphite optional dependencies

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-1550

These dependencies must now be installed manually by users. Updated docs and the LICENSE for Gremlin Server. Gets rid of the weird one-off exclusion of an LGPL license (couldn't support Ganglia out of the box anyway) and lessens transitive dependency burdens.

Builds with `mvn clean install && mvn verify -pl gremlin-server -DskipIntegrationTests=false && mvn verify -pl gremlin-console -DskipIntegrationTests=false`

VOTE +1

### CODE REVIEW

File: gremlin-server/pom.xml [hunk 1]
```diff
@@ -63,10 +63,12 @@ limitations under the License.
         <dependency>
             <groupId>com.codahale.metrics</groupId>
             <artifactId>metrics-graphite</artifactId>
+            <optional>true</optional>
```
Comment (dkuppitz) [inline gremlin-server/pom.xml]: Shouldn't this include the version? Asking because you've explicitly mentioned 3.0.2 before.

File: pom.xml [hunk 2]
```diff
@@ -597,6 +597,7 @@ limitations under the License.
                 <groupId>com.codahale.metrics</groupId>
                 <artifactId>metrics-graphite</artifactId>
                 <version>${metrics.version}</version>
+                <optional>true</optional>
```
Comment (dkuppitz) [inline pom.xml]: Same here. It's probably better to have the version tag included.

File: gremlin-server/pom.xml (hunk 1 above)
Comment (spmallette) [inline gremlin-server/pom.xml]: All of metrics is under `<dependencyManagement>` in the root pom.xml so the version is defined there:

https://github.com/apache/tinkerpop/blob/268333b4778c87ce6c2429eda97e3e1adc7757e3/pom.xml#L602-L642

I don't think it needs to be specified again.

File: pom.xml (hunk 2 above)
Comment (spmallette) [inline pom.xml]: I don't follow - the `<version>` is there. It's presented as a pom variable, but it's defined.

Comment (dkuppitz) [inline pom.xml]: Doh, I probably only focused on the addition of the optional tag and assumed it's the same dependency block as above. All good here.

**Comment by dkuppitz** [comment]
VOTE: +1

**Comment by okram** [comment]
VOTE +1

**Pull request merged**